includes/DbService.php

```diff
@@ -97,8 +97,6 @@ public function dropFilterValuesTempTable() {
 	 * set of filters and either a new subcategory or a new filter.
 	 */
 	public function getNumResults( $subcategory, $subcategories, $new_filter = null ) {
-		// Escape the given values to prevent SQL injection
-		$subcategory = $this->dbr->addQuotes( $subcategory );
 		foreach ( $subcategories as $key => $value ) {
 			$subcategories[$key] = $this->dbr->addQuotes( $value );
```

Comment:
Shouldn't we remove this one, too?

Reply:
hello Markus, yes `addQuotes` in the method `getNumResults` is not required since another `addQuotes` is applied to the same variable $subcategory, in the function `SqlProvider::getSQLFromClauseForCategory` which is called right below, see [here](https://github.com/SemanticMediaWiki/SemanticDrilldown/blob/master/includes/DbService.php) and [here](https://github.com/SemanticMediaWiki/SemanticDrilldown/blob/master/includes/Sql/SqlProvider.php)

Reply:
That's right, but isn't the same true for the $subcategories? So, adding quotes to them here should be removed, too. This is what I asked.

Reply:
ah, I misunderstood the question.
Yes, actually all this loop should be removed

```
		foreach ( $subcategories as $key => $value ) {
			$subcategories[$key] = $this->dbr->addQuotes( $value );
		}
```

(in the function getNumResults)